swig/cpp/src/Xpath.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file Xpath.h
+ * @author Mislav Novakovic <mislav.novakovic@sartura.hr>
+ * @brief Sysrepo class header for C header xpath_utils.h.
+ *
+ * @copyright
+ * Copyright 2016 Deutsche Telekom AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef XPATH_H
+#define XPATH_H
+
+#include <iostream>
+
+extern "C" {
+#include "sysrepo/xpath_utils.h"
+}
+
+using namespace std;
+
+class Xpath_Ctx:public Throw_Exception
+{
+public:
+    Xpath_Ctx();
+    Xpath_Ctx(sr_xpath_ctx_t *state);
+    ~Xpath_Ctx();
+    char *next_node(char *xpath) {return sr_xpath_next_node(xpath, _state);};
```

Comment:
I think it would be nice if xpath arguments had default value = NULL and even better if it xpath was passed to constructor and then to the first call of Xpath_Ctx instance functions. 
```c++
Xpath_Ctx(char *xpath);
char *next_node() {return sr_xpath_next_node(get_xpath_internal(), _state);};
private:
char *get_xpath_internal() /* returns xpath provided to constructur for the first call, NULL for subsequent */
```
This would apply to all functions that has `xpath` and `state` argument.